### PR TITLE
Unnest the Timer class

### PR DESCRIPTION
**Pull request by alex**
classes are relatively expensive to create, and particularly disadvantageous on PyPy

A benchmark could be created if this was useful

### CODE REVIEW

**Comment by brian-brazil** [comment]
There's two other Timer classes.

How big is the performance difference?

**Comment by alex** [comment]
Using this simple benchmark:

```py
from prometheus_client import Histogram


def main():
    h = Histogram("nonsense_latency_seconds", "Latency of nonsense (seconds)")

    for i in xrange(100000):
        with h.time():
            pass


if __name__ == "__main__":
    main()
```

On PyPy 5.1.1

*WIth patch:*

```
(tempenv-43511495362f0) ~ $ time python t.py
        0.22 real         0.17 user         0.04 sys
(tempenv-43511495362f0) ~ $ time python t.py
        0.23 real         0.18 user         0.04 sys
(tempenv-43511495362f0) ~ $ time python t.py
        0.23 real         0.18 user         0.05 sys
```

*master:*
```
(tempenv-43511495362f0) ~ $ time python t.py
        4.45 real         4.29 user         0.10 sys
(tempenv-43511495362f0) ~ $ time python t.py
        4.50 real         4.30 user         0.11 sys
(tempenv-43511495362f0) ~ $ time python t.py
        4.50 real         4.31 user         0.12 sys
```

So pretty massive. This optimization should be made to all the other nested `Timer` classes.

**Comment by brian-brazil** [comment]
That's a pretty nice improvement, most of the slowness is due to the mutex at that point.

**Comment by alex** [comment]
Indeed. Do you want me to move the other classes in this PR, or a separate one?

**Comment by brian-brazil** [comment]
All in one PR please.

**Comment by alex** [comment]
@brian-brazil done.

File: prometheus_client/core.py [hunk 1]
```diff
@@ -399,25 +399,7 @@ def count_exceptions(self, exception=Exception):
         type is raised up out of the code.
         '''
```
Comment (brian-brazil) [inline prometheus_client/core.py]: We can remove this blank line now.

**Comment by brian-brazil** [comment]
That looks generally fine, can you squash your commits please?

**Comment by alex** [comment]
@brian-brazil done!

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**